### ISSUE TITLE
RGB image to PALETTE

### ISSUE THREAD

**Issue by paulopires16** [python-pillow/Pillow#2994]
I'm trying to convert RGB image to PALETTE image with PIL: 
img = Image.open(tiffile) 
img = img1convert("P", palette=Image.ADAPTIVE, colors=256) 
img.save(newtiffile+"_index.tif")     
img.close()
...but, for example, the white color is 245, not 255 :(

**Comment by wiredfool** [comment]
Yes. The palette is a lookup table. In general, there is no relationship between the index of a palette entry and it's value.

**Comment by paulopires16** [comment]
So, there's not possible a conversion guarantee the colors?

**Comment by wiredfool** [comment]
You can convert the image to match an existing palette.

**Comment by paulopires16** [comment]
Thanks. 
How I can create that palette?

**Comment by radarhere** [comment]
- 'Adaptive Palette?' - The 'palette' argument can be either `Image.WEB` (default) or `Image.ADAPTIVE`. If it is adaptive, it is converted to use the number of colors that you provided in the 'colors' argument. Web refers to a palette of 216 colors.
- 'download?' - I don't know what you're referring to.
- 'how I can use with pillow?' - I would suggest looking at the [getpalette](https://pillow.readthedocs.io/en/5.3.x/reference/Image.html#PIL.Image.Image.getpalette) and [putpalette](https://pillow.readthedocs.io/en/5.3.x/reference/Image.html#PIL.Image.Image.putpalette) methods. That should allow you to get a list of RGB values from an image, and set it for an image -

```
>>> p = im.getpalette()
>>> print(p)
[255, 253, 252, 251, 253, 253, ...]
>>> im.putpalette(p)
```
